src/zerocoin_v3.cpp

```diff
@@ -637,23 +634,27 @@ bool CSigmaState::AddMintsToStateAndBlockIndex(
 
                 coinGroup.lastBlock = index;
             }
+            coinGroup.nCoins += mintsWithThisDenom.size();
         }
         else {
             latestCoinIds[denomination] = ++mintCoinGroupId;
             SigmaCoinGroupInfo& newCoinGroup = coinGroups[std::make_pair(denomination, mintCoinGroupId)];
             newCoinGroup.firstBlock = newCoinGroup.lastBlock = index;
-            newCoinGroup.nCoins = 1;
+            newCoinGroup.nCoins = mintsWithThisDenom.size();
         }
-        CMintedCoinInfo coinInfo;
-        coinInfo.denomination = denomination;
-        coinInfo.id = mintCoinGroupId;
-        coinInfo.nHeight = index->nHeight;
-        mintedPubCoins.insert(std::make_pair(pubCoin, coinInfo));
 
-        LogPrintf("AddMints: mint added denomination=%d, id=%d\n", denomination, mintCoinGroupId);
-        index->sigmaMintedPubCoins[{denomination, mintCoinGroupId}].push_back(pubCoin);
+        for (const auto& mint : mintsWithThisDenom) {
+            CMintedCoinInfo coinInfo;
+            coinInfo.denomination = denomination;
+            coinInfo.id = mintCoinGroupId;
+            coinInfo.nHeight = index->nHeight;
+            mintedPubCoins.insert(std::make_pair(mint, coinInfo));
+
+            LogPrintf("AddMintsToStateAndBlockIndex: mint added denomination=%d, id=%d\n", denomination, mintCoinGroupId);
+            index->sigmaMintedPubCoins[{denomination, mintCoinGroupId}].push_back(mint);
+        }
     }
-    return 1;
+    return true;
```

Comment:
Seem like we don't need a returned value anymore. So change this method to `void` instead will be better.

Reply:
make sense.